advanced/Exercise_8.md

```diff
@@ -0,0 +1,138 @@
+# Exercise 8 - Using git subtree
+
+This exercise is designed to be paired with Exercise 6 to help you compare and contrast the two main methods for embedding a git repository into another git repository. In this exercise, we will learn to use git subtree to nest a git repository inside another one. We will first add the git subtree and examine how the host and nested repositories deal with changes in their respective files. We will then learn how to pull and push changes to and from the sub-repository.
+
+This exercise uses the same git repository that was created in Exercise 3. If you have not already done so, you can create it by following the instructions in the `Initialize the git repository` section [here](./Exercise_3.md).
+
+* [Add the posters repository using git subtree](#subtree)
+
+* [Examine how git deals with changes to both repositories](#examine)
+
+* [Create new content locally and update the sub-repository](#push)
+
+* [Create new content in the sub-repository and update the local repository](#pull)
+
+## Add the posters repository using git subtree <a name="subtree"></a>
+
+Navigate to the folder containing the `conference_planning` folder; in other words you should be in a folder that does not contain a git repository. Make a copy of the conference planning repository to use for adding the subtree.
+
+```plaintext
+cp -r conference_planning conference_subtree
+```
+
+Let's add your `posters` Github fork we created in Exercise 6 as a subtree to our conference planning repository.
+
+```plaintext
+cd conference_subtree
+git subtree add --prefix posters YOUR_FORK_ADDRESS main --squash
```

Comment:
I got the error "Working treee has modifications. Cannot add." and have no idea why. My working tree is clean. Maybe we need to look into this as I can imagine others will run into the same problem?
(`git reset --hard` fixed it)